File: datasets/C++/Copilot/DataStructures/hard4.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cout << "Enter number of elements: ";
    cin >> N;
    
    vector<int> A(N);
    cout << "Enter the elements: ";
    for (int i = 0; i < N; ++i) cin >> A[i];

    int max_si = 0;
    for (int i = 0; i < N - 1; ++i) {
        int m1 = min(A[i], A[i+1]);
        int m2 = max(A[i], A[i+1]);
        int si = ((m1 & m2) ^ (m1 | m2)) & (m1 ^ m2);
        max_si = max(max_si, si);
    }

    cout << "Maximum Si: " << max_si << endl;
    return 0;
}

File: datasets/Python/Copilot/DataStructures/hard4.py
# Manual Input
N = int(input("Enter number of elements: "))
A = list(map(int, input("Enter the elements: ").split()))

def max_si(arr):
    n = len(arr)
    max_si_val = 0
    for i in range(n - 1):
        m1, m2 = arr[i], arr[i+1]
        if m1 > m2:
            m1, m2 = m2, m1
        si = ((m1 & m2) ^ (m1 | m2)) & (m1 ^ m2)
        max_si_val = max(max_si_val, si)
    return max_si_val

print("Maximum Si:", max_si(A))